Utilities/DataLoading/EdFi.LoadTools.Test/SmokeTests/APIGetTests.cs

```diff
@@ -75,78 +81,99 @@ public class ApiGetTests
+                                                JsonConvert.SerializeObject(
+                                                    _data.SingleOrDefault(x => x[IdName].Value<int>() == id)));
+                                        }
+                                        else if (!string.IsNullOrEmpty(context.Request.Query["offset"]) &&
+                                                 !string.IsNullOrEmpty(context.Request.Query["limit"]))
+                                        {
+                                            var skip = int.Parse(context.Request.Query["offset"]);
+                                            var take = int.Parse(context.Request.Query["limit"]);
+
+                                            await context.Response.WriteAsync(
+                                                JsonConvert.SerializeObject(_data.Skip(skip).Take(take)));
+                                        }
+                                        else if (!string.IsNullOrEmpty(context.Request.Query[KeyName]))
+                                        {
+                                            var keyValue = context.Request.Query[KeyName];
+
+                                            await context.Response.WriteAsync(
+                                                JsonConvert.SerializeObject(
+                                                    _data.SingleOrDefault(
+                                                        x =>
+                                                            x[KeyName].Value<string>() == keyValue)));
+                                        }
+                                        else if (!string.IsNullOrEmpty(context.Request.Query[PropertyName]))
+                                        {
+                                            var propertyValue = context.Request.Query[PropertyName];
+
+                                            await context.Response.WriteAsync(
+                                                JsonConvert.SerializeObject(
+                                                    _data.Where(
+                                                        x =>
+                                                            x[PropertyName].Value<string>() == propertyValue)));
+                                        }
+                                        else
+                                        {
+                                            await context.Response.WriteAsync(JsonConvert.SerializeObject(_data));
+                                        }
+                                    }));
+                        })
+                ;
+
+            Host = hostBuilder.Build();
+            Host.StartAsync();
         }
 
         [OneTimeTearDown]
         public void Cleanup()
         {
-            _webApp.Dispose();
+            Host.StopAsync();
+            Host.Dispose();
         }
 
         [Test]
         public void GetAll_should_store_results_in_dictionary()
```

Comment:
these tests should be async Task as the type

Reply:
Updated